src/waveform/renderers/waveformmarkrange.cpp

```diff
@@ -61,6 +46,37 @@ WaveformMarkRange::WaveformMarkRange(
     } else {
         m_durationTextLocation = DurationTextLocation::After;
     }
+
+    if (!m_activeColor.isValid()) {
+        //vRince kind of legacy fallback ...
+        // As a fallback, grab the mark color from the parent's MarkerColor
+        QString rangeName = startControl;
+        if (startControl.contains(QStringLiteral("_start_position"))) {
+            rangeName = startControl.remove(QStringLiteral("_start_position"));
+        }
+        m_activeColor = signalColors.getAxesColor();
+        qDebug() << "Didn't get Color for mark range" << rangeName
+                << "- using parent's AxesColor:" << m_activeColor;
+    } else {
+        m_activeColor = WSkinColor::getCorrectColor(m_activeColor);
+    }
+
+    if (!m_disabledColor.isValid()) {
+        if (enabledControl.isEmpty()) {
+            m_disabledColor = QColor(Qt::transparent);
+        } else {
+            // Show warning only when there's no EnabledControl,
+            // like for intro & outro ranges.
+            QString rangeName = startControl;
+            if (startControl.contains(QStringLiteral("_start_position"))) {
+                rangeName = startControl.remove(QStringLiteral("_start_position"));
```

Comment:
you can just call 
rangeName = startControl.remove(QStringLiteral("_start_position"));
the contais() check is done implicite and the if statement above redundant.

Reply:
working on it..